iocore/net/quic/QUICPinger.cc

```diff
@@ -69,9 +73,16 @@ QUICPinger::generate_frame(uint8_t *buf, QUICEncryptionLevel level, uint64_t /*
 
   if (this->_need_to_fire[static_cast<int>(level)] > 0 && maximum_frame_size > 0) {
     // don't care ping frame lost or acked
-    frame                                        = QUICFrameFactory::create_ping_frame(buf, 0, nullptr);
-    this->_need_to_fire[static_cast<int>(level)] = 0;
+    frame = QUICFrameFactory::create_ping_frame(buf, 0, nullptr);
+    --this->_need_to_fire[static_cast<int>(level)];
   }
```

Comment:
Decrease count when generate ping frame. Because we only generate one ping frame per packet. See `packetize_frames`. So the count of `_need_to_fire ` infer to how many packets we need to send out .